discovery/consul/consul.go

```diff
@@ -481,48 +484,48 @@ func (srv *consulService) watch(ctx context.Context, ch chan<- []*targetgroup.Gr
 	tgroup := targetgroup.Group{
 		Source:  srv.name,
 		Labels:  srv.labels,
-		Targets: make([]model.LabelSet, 0, len(nodes)),
+		Targets: make([]model.LabelSet, 0, len(serviceNodes)),
 	}
 
-	for _, node := range nodes {
-
+	for _, serviceNode := range serviceNodes {
 		// We surround the separated list with the separator as well. This way regular expressions
 		// in relabeling rules don't have to consider tag positions.
-		var tags = srv.tagSeparator + strings.Join(node.ServiceTags, srv.tagSeparator) + srv.tagSeparator
+		var tags = srv.tagSeparator + strings.Join(serviceNode.Service.Tags, srv.tagSeparator) + srv.tagSeparator
 
 		// If the service address is not empty it should be used instead of the node address
 		// since the service may be registered remotely through a different node.
 		var addr string
-		if node.ServiceAddress != "" {
-			addr = net.JoinHostPort(node.ServiceAddress, fmt.Sprintf("%d", node.ServicePort))
+		if serviceNode.Service.Address != "" {
+			addr = net.JoinHostPort(serviceNode.Service.Address, fmt.Sprintf("%d", serviceNode.Service.Port))
 		} else {
-			addr = net.JoinHostPort(node.Address, fmt.Sprintf("%d", node.ServicePort))
+			addr = net.JoinHostPort(serviceNode.Node.Address, fmt.Sprintf("%d", serviceNode.Service.Port))
 		}
 
 		labels := model.LabelSet{
 			model.AddressLabel:  model.LabelValue(addr),
-			addressLabel:        model.LabelValue(node.Address),
-			nodeLabel:           model.LabelValue(node.Node),
+			addressLabel:        model.LabelValue(serviceNode.Node.Address),
+			nodeLabel:           model.LabelValue(serviceNode.Node.Node),
 			tagsLabel:           model.LabelValue(tags),
-			serviceAddressLabel: model.LabelValue(node.ServiceAddress),
-			servicePortLabel:    model.LabelValue(strconv.Itoa(node.ServicePort)),
-			serviceIDLabel:      model.LabelValue(node.ServiceID),
+			serviceAddressLabel: model.LabelValue(serviceNode.Service.Address),
+			servicePortLabel:    model.LabelValue(strconv.Itoa(serviceNode.Service.Port)),
+			serviceIDLabel:      model.LabelValue(serviceNode.Service.ID),
+			healthLabel:         model.LabelValue(serviceNode.Checks.AggregatedStatus()),
 		}
 
 		// Add all key/value pairs from the node's metadata as their own labels.
-		for k, v := range node.NodeMeta {
+		for k, v := range serviceNode.Node.Meta {
 			name := strutil.SanitizeLabelName(k)
 			labels[metaDataLabel+model.LabelName(name)] = model.LabelValue(v)
 		}
 
 		// Add all key/value pairs from the service's metadata as their own labels.
-		for k, v := range node.ServiceMeta {
+		for k, v := range serviceNode.Service.Meta {
 			name := strutil.SanitizeLabelName(k)
 			labels[serviceMetaDataLabel+model.LabelName(name)] = model.LabelValue(v)
 		}
 
 		// Add all key/value pairs from the service's tagged addresses as their own labels.
-		for k, v := range node.TaggedAddresses {
+		for k, v := range serviceNode.Node.TaggedAddresses {
```

Comment:
This doesn't seem to exist for the Health call

Reply:
Hmm, it does : https://www.consul.io/api/health.html#sample-response-2

Reply:
Ah, I was looking in the Service rather than the Node